components/base/kinematicbase/ptgKinematics.go

```diff
@@ -224,33 +222,34 @@ func (ptgk *ptgBaseKinematics) goToInputs(ctx context.Context, inputs []referenc
 			nil,
 		)
 		if err != nil {
-			stopCtx, cancelFn := context.WithTimeout(context.Background(), time.Second*5)
-			defer cancelFn()
-			return multierr.Combine(err, ptgk.Base.Stop(stopCtx, nil))
+			return ptgk.stopMotion(err)
 		}
-		utils.PanicCapturingGo(func() {
-			// We need to update currentInputs as we move through the arc.
-			for timeElapsed := 0.; timeElapsed <= step.timestepSeconds; timeElapsed += inputUpdateStep {
-				distIncVel := step.linVelMMps.Y
-				if distIncVel == 0 {
-					distIncVel = step.angVelDegps.Z
-				}
-				ptgk.inputLock.Lock()
-				ptgk.currentInput = []referenceframe.Input{inputs[0], inputs[1], {math.Abs(distIncVel) * timeElapsed}}
-				ptgk.inputLock.Unlock()
-				utils.SelectContextOrWait(ctx, time.Duration(inputUpdateStep*1000*1000)*time.Microsecond)
-			}
-		})
 
-		if !utils.SelectContextOrWait(ctx, timestep) {
-			ptgk.logger.CDebug(ctx, ctx.Err().Error())
-			// context cancelled
-			break
+		// We need to update currentInputs as we move through the arc.
+		for timeElapsed := 0.; timeElapsed <= step.timestepSeconds; timeElapsed += inputUpdateStep {
+			if ctx.Err() != nil {
+				return ptgk.stopMotion(err)
```

Comment:
@biotinker I added this function (`ptgk.stopMotion`) instead of doing
``` 
stopCtx, cancelFn := context.WithTimeout(context.Background(), time.Second*5)
defer cancelFn()
return multierr.Combine(err, ptgk.Base.Stop(stopCtx, nil))
```

Doing the above resulted in `defer within an immediate outer for usage found "defer cancelFn()"` from the linter. 
A google search of this string led me to the following web page (https://github.com/chromedp/chromedp/issues/746#issuecomment-771388870)

Hence, we introduced `ptgk.stopMotion(err error)` as a method to circumvent this linter issue.

------------------------------

May you please confirm that you agree with this choice?

Reply:
1) rather than making `err` a parameter to `stopMotion`, let's just have it return the err from `Stop`. The caller can be responsible for combining, i.e. `return multierr.Combine(err, ptgk.stopMotion())`

2) rather than adding this as a method for `ptgk`, let's just make it an inline function within `goToInputs`.